src/main/java/org/opensearch/ad/model/DetectorValidationIssue.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package org.opensearch.ad.model;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+
+import com.google.common.base.Objects;
+
+/**
+ * DetectorValidationIssue is a single validation issue found for detector.
+ *
+ * For example, if detector's multiple features are using wrong type field or non existing field
+ * the issue would be in `detector` aspect, not `model`;
+ * and its type is FEATURE_ATTRIBUTES, because it is related to feature;
+ * message would be the message from thrown exception;
+ * subIssues are issues for each feature;
+ * suggestion is how to fix the issue/subIssues found
+ */
+public class DetectorValidationIssue implements ToXContentObject, Writeable {
+    private static final String MESSAGE_FIELD = "message";
+    private static final String SUGGESTED_FIELD_NAME = "suggested_value";
+    private static final String SUB_ISSUES_FIELD_NAME = "sub_issues";
+
+    private final ValidationAspect aspect;
+    private final DetectorValidationIssueType type;
```

Comment:
Minor thought - is it necessary to have a separate class file for the type? Can it be implemented directly as an enum in this class for simplicity?

Reply:
I think I want to keep this as a separate class file as it can potentially be changed and it keeps consistent with a lot of other enum classe files that were made.

Reply:
Sure, I'm ok to keep it separate.